backend/metrics/metrics_test.go

```diff
@@ -0,0 +1,56 @@
+package metrics
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	managedgitopsv1alpha1 "github.com/redhat-appstudio/managed-gitops/backend-shared/apis/managed-gitops/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+var _ = Describe("Test for Gitopsdeployment metrics counter", func() {
+
+	Context("Prometheus metrics responds to count of active/failed GitopsDeployments", func() {
+		It("tests Add/Update, Remove, SetError function on a gitops deployment", func() {
+
+			ClearMetrics()
+
+			numberOfGitOpsDeploymentsInErrorState := testutil.ToFloat64(GitopsdeplFailures)
+			totalNumberOfGitOpsDeploymentMetrics := testutil.ToFloat64(Gitopsdepl)
+
+			gitopsDepl := &managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-gitops-depl",
+					Namespace: "gitops-depl-namespace",
+					UID:       uuid.NewUUID(),
+				},
+			}
+
+			AddOrUpdateGitOpsDeployment(gitopsDepl.Name, gitopsDepl.Namespace, string(gitopsDepl.UID))
```

Comment:
Can you add `By("...")` statements after each action in this function, so that the test print out what it's doing after each step?

For example, `By("simulating a new valid GitOpsDeployment being processed")`